services/api/service.go

```diff
@@ -950,15 +1102,43 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}
 
-	builderIsHighPrio, builderIsBlacklisted, err := api.redis.GetBlockBuilderStatus(payload.Message.BuilderPubkey.String())
+	builderStatus, err := api.redis.GetBlockBuilderStatus(payload.Message.BuilderPubkey.String())
 	log = log.WithFields(logrus.Fields{
-		"builderIsHighPrio":    builderIsHighPrio,
-		"builderIsBlacklisted": builderIsBlacklisted,
+		"builderStatus": builderStatus.String(),
 	})
 	if err != nil {
 		log.WithError(err).Error("could not get block builder status")
 	}
 
+	if builderStatus == common.Blacklisted {
```

Comment:
returning the blacklist info here would allow builders to detect whether they're blacklisted by sending an incorrect timestamp. seems better to do the primary checks first 🤔

Reply:
There's an easy strategy to detect blacklisting. Just submit an invalid block and see if the relay returns `StatusOK`. The intent of this change is to minimise the resources consumed by blacklisted builders.

Reply:
That is true, good point :)

Reply:
I would also advocate for removing `time.Sleep(200 * time.Millisecond)` and return an error message.

Reply:
done! now calling 

```
api.RespondError(w, http.StatusForbidden, fmt.Sprintf("builder with pubkey %x is blacklisted", payload.Message.BuilderPubkey.String()))
```

Reply:
looks good to me! :)

Reply:
I don't like that change, it makes it too obvious to a builder that they are blacklisted, and they will just rotate their key. i'm for keeping both the timeout and just the OK status response as previously. also don't really see a downside to that.

Reply:
@michaelneuder: I suggest fully reverting all the changes to the blacklisting logic. It was my bad—it should be different discussion/PR. That will also reduce the diff and make the PR easier to review :)

Reply:
reverted!